### DISCUSSION TITLE
voice channel connect doesnt work and i dont know why

### DISCUSSION THREAD

**Discussion by ghost** [Rapptz/discord.py#9210] [Q&A]
the code is listed below
```
import discord, youtube_dl
from discord.ext import commands


class Music(commands.Cog):
    def __init__(self, bot):
        self.bot = bot

    @commands.command(pass_context=True)
    async def join(self, ctx):
        vc = ctx.message.author.voice.channel
        await vc.connect()

async def setup(bot):
    await bot.add_cog(Music(bot))
    print("[Console] music.py loaded up!")
```
cog itself works, command "join" doesnt. why?

**Comment by tookender** [comment]
not an answer but a question, why do you have `pass_context`? that is very outdated and has not existed for years (since the rewrite of discord.py)
looks like you might be following an outdated forum post or tutorial.

**Reply by ghost** [reply]
> not an answer but a question, why do you have `pass_context`? that is very outdated and has not existed for years (since the rewrite of discord.py) looks like you might be following an outdated forum post or tutorial.

idk found old stackquestion once, but i haven't found working one in new

**Comment by du-cki** [comment]
Hello. Can you elaborate on "doesn't work"? I couldn't repro the issue. Do you get errors? If not do you have an error handler that could potentially eat up the errors?

P.S. I've just noticed this post is 22 days old. If your issue has been resolved, please close this discussion.
